Update creating-a-github-pages-site-with-jekyll.md (#13938)
github/docs: content/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll.md

 If you chose to publish your site from the `gh-pages` branch, create and checkout the `gh-pages` branch.
 ```shell
 $ git checkout --orphan gh-pages

>  # Creates a new branch with no history, called gh-pages and switches to the gh-pages branch
>  # If your repository had content and you want to remove it from the gh-pages branch, do
>  $ git rm -rf .

Reviewer: `git checkout --orphan` does create a branch with no contents. You might be seeing the files in your working directory (not yet committed to your branch) and staging area.

```suggestion
 # Creates a new branch, with no history or contents, called gh-pages, and switches to the gh-pages branch
 $ git rm -rf 
 # Removes the contents from your default branch from the working directory
```
